Release/tests/functional/http/client/client_construction.cpp

```diff
@@ -173,7 +173,7 @@ SUITE(client_construction)
         VERIFY_ARE_EQUAL(baseclient2.base_uri(), m_uri);
     }
 
-#if defined(CPPREST_FORCE_HTTP_LISTENER_ASIO)
+#if !defined(_WIN32) && !defined(__cplusplus_winrt) || defined(CPPREST_FORCE_HTTP_CLIENT_ASIO)
```

Comment:
Can this condition be factored out into a separate named setting that describes it? It seems like we should be already setting CPPREST_FORCE_HTTP_CLIENT_ASIO on non-Windows platforms?

Reply:
The CMake config does I think, but looking at how both this and the similar `LISTENER` preprocessor symbol are used throughout the code, the assumption in almost every case is that the `!defined(_WIN32)` check is the primary condition since WinHTTP/HttpSys can't be used, and `FORCE` is only applied on Windows.

Reply:
@BillyONeal Please would you let me know what you would like to be addressed before merging this?

[Searching the project for CPPREST_FORCE_HTTP_CLIENT_ASIO](https://github.com/microsoft/cpprestsdk/search?q=defined%28CPPREST_FORCE_HTTP_CLIENT_ASIO%29) and [for CPPREST_FORCE_HTTP_LISTENER_ASIO](https://github.com/microsoft/cpprestsdk/search?q=defined%28CPPREST_FORCE_HTTP_LISTENER_ASIO%29) show that in other places, these conditions are used as I've suggested here.

Thanks!